x-pack/plugins/ml/public/application/datavisualizer/file_based/components/fields_stats_grid/filter_fields.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FileBasedFieldVisConfig } from '../../../stats_table/types';
+import { ML_JOB_FIELD_TYPES } from '../../../../../../common/constants/field_types';
+
+export function filterFields(
+  fields: Array<
+    | FileBasedFieldVisConfig
+    | {
+        fieldName: string;
+        type: 'text' | 'unknown';
+        stats: { mean: number; count: number; sampleCount: number; cardinality: number };
+      }
```

Comment:
worth extracting to an interface

Reply:
Updated here https://github.com/elastic/kibana/pull/87598/commits/93d64ccbcc8659622005d9447e3371942c3fef40